### PR TITLE
Support listing all files in a dataset independent of folder hierarchy in the dataset & table update fix to 'develop' 

### PR DESCRIPTION
**Pull request by milen-sage**
This keeps abreast with #301 and #299

### CODE REVIEW

File: schematic/synapse/store.py [hunk 1]
```diff
@@ -430,15 +453,20 @@ def associateMetadataWithFiles(self, metadataManifestPath: str, datasetId: str)
...
+            manifest = update_df(existing_table, manifest, 'entityId')
+
+            self.syn.store(Table(existingTableId, manifest, etag = existing_results.etag))
+            # remove system metadata from manifest
+            manifest.drop(columns = ['ROW_ID', 'ROW_VERSION'], inplace = True)
+            
+        else:  
+            # create table using latest manifest content
+            table = build_table(datasetName + "_table", datasetParentProject, manifest)
+            table = self.syn.store(table)
```
Comment (sujaypatil96) [inline schematic/synapse/store.py:468]: Bug fix 2: Tested the updation of existing table associated with a given dataset. Works well functionally.

**Review by sujaypatil96 — APPROVED** [review]
Just left a few trivial notes @milen-sage. Functionally, the PR is good to merge.👍

File: schematic/synapse/store.py [hunk 2]
```diff
@@ -338,6 +346,20 @@ def getAllManifests(self) -> List[str]:
...
+        Args:
+            synapse_id: synapse ID of the table to query
+        """
+
+        results = self.syn.tableQuery("SELECT * FROM %s" % synapse_id)
+        df = results.asDataFrame(rowIdAndVersionInIndex = False)
+
+        return df, results
+
+
```
[the remark this answers is not in the document]
Comment (milen-sage) [reply schematic/synapse/store.py]: See latest commit; which follows this suggestion.

File: schematic/synapse/store.py [hunk 3]
```diff
@@ -196,39 +197,46 @@ def getStorageDatasetsInProject(self, projectId: str) -> List[str]:
...
-        return sorted_files_list
-        
+                    if fullpath:
+                        # append directory path to filename
+                        filename  = (dirpath[0] + "/" + filename[0], filename[1])
+
+                    # add file name file id tuple, rearranged so that id is first and name follows
+                    file_list.append(filename[::-1])
+ 
+        return file_list
```
[the remark this answers is not in the document]
Comment (milen-sage) [reply schematic/synapse/store.py]: I agree line 226 is hard to read. I left a TODO note in the latest commit code. 

Let's merge as is for now. Let's think a bit more how to make the logic more maintainable/readable. I agree your proposal splits up the logic in more natural way, but let's see if we can compact the if/else branches a bit and issue a separate PR. We can discuss during our meeting.

**Pull request merged**